yarn-project/simulator/src/avm/avm_gas.ts

```diff
@@ -102,12 +122,29 @@ export const GasCosts = {
   [Opcode.PEDERSEN]: TemporaryDefaultGasCost, // temp - may be removed, but alot of contracts rely on i: TemporaryDefaultGasCost,t
 } as const;
 
+/** Returns the fixed gas cost for a given opcode, or throws if set to dynamic. */
+export function getFixedGasCost(opcode: Opcode): Gas {
+  const cost = GasCosts[opcode];
+  if (cost === DynamicGasCost) {
+    throw new Error(`Opcode ${Opcode[opcode]} has dynamic gas cost`);
+  }
+  return cost;
+}
+
+/** Returns the additional cost from indirect accesses to memory. */
+export function getCostFromIndirectAccess(indirect: number): Partial<Gas> {
+  const indirectCount = Addressing.fromWire(indirect).modePerOperand.filter(
+    mode => mode === AddressingMode.INDIRECT,
+  ).length;
+  return { l2Gas: indirectCount * GasCostConstants.COST_PER_INDIRECT_ACCESS };
+}
+
 /** Constants used in base cost calculations. */
 export const GasCostConstants = {
   SET_COST_PER_BYTE: 100,
   CALLDATACOPY_COST_PER_BYTE: 10,
   ARITHMETIC_COST_PER_BYTE: 10,
-  ARITHMETIC_COST_PER_INDIRECT_ACCESS: 5,
+  COST_PER_INDIRECT_ACCESS: 5,
```

Comment:
Why is this not a COST_PER_MEMORY_ACCESS? I'd expect that any memory access (direct or indirect) costs the same?

I.e., same as SET_COST_PER_BYTE as well? CALLDATACOPY also just copies bytes from/to mem (but I could see why it might be a special case).

Reply:
The yellow paper states that [indirect accesses have an overhead](https://yp-aztec.netlify.app/docs/public-vm/execution#gas-cost-notes-and-examples):

> An instruction with "offset" arguments (like ADD and many others), has increased cost for each offset argument that is flagged as "indirect".

Agree on `SET` though. I'll review that, along with memory accesses in general, in a subsequent PR.